src/sentry/notifications/utils/__init__.py

```diff
@@ -129,6 +130,30 @@ def summarize_issues(
     return rv
 
 
+def get_email_link_extra_params(
+    referrer: str = "alert_email",
+    rule_details: Sequence[NotificationRuleDetails] | None = None,
+    alert_timestamp: int | None = None,
+) -> dict[int, str]:
+    alert_timestamp_str = (
+        str(round(time.time() * 1000)) if not alert_timestamp else str(alert_timestamp)
+    )
+    return {
+        rule_detail.id: "?"
+        + str(
+            urlencode(
+                {
+                    "referrer": referrer,
+                    "alert_type": str(AlertRuleTriggerAction.Type.EMAIL.name).lower(),
+                    "alert_timestamp": alert_timestamp_str,
+                    "alert_rule_id": rule_detail.id,
+                }
+            )
+        )
+        for rule_detail in rule_details
+    }
+
+
 def get_group_settings_link(
```

Comment:
Missed this in the previous pr but this could probably also do with a test.

Reply:
https://github.com/getsentry/sentry/blob/565903080c45097a321c028a027dc882fb4c046c/tests/sentry/notifications/test_utils.py#L187